fasta/faBin.go

```diff
@@ -0,0 +1,250 @@
+package fasta
+
+import (
+	"github.com/vertgenlab/gonomics/dna"
+	"log"
+)
+
+//BinGenomeNoBreaks takes in an entire genome which is sorted largest to smallest contig and breaks up the fasta so
+//that smaller contigs get combined into a single fasta, while large contigs become a single fasta on their own. The user
+//must specify the number of bins for the genome to be broken into, the genome must have more contigs than bins in order
+//to combine any contigs and equal number bins to contigs if each contig gets its own record. The bins will all be
+//filled with the first contig encountered when it's empty, and then the smallest of those bins will be filled when the
+//contig is equal to binNum+1. The minSize option allows for a user to specify a minimum length of sequence to go into
+//each bin and in this case the number of bins returned depends on the minSize and the binNum will be ignored.
+func BinGenomeNoBreaks(genome []Fasta, binNum int, minSize int) map[int][]Fasta {
+	var bins map[int][]Fasta
+
+	if minSize != -1 {
+		bins = binMinSize(genome, minSize)
+	} else {
+		bins = make(map[int][]Fasta, binNum)
+		if len(genome) > binNum {
+			for n := 0; n < binNum; n++ {
+				bins[n] = append(bins[n], genome[n])
+			}
+			bins = fillSmallestBin(bins, genome)
+		} else if len(genome) == binNum {
+			for n := 0; n < binNum; n++ {
+				bins[n] = append(bins[n], genome[n])
+			}
+		} else {
+			log.Fatal("Number of bins is greater than the number of contigs in the given genome. Reduce bin number.")
+		}
+	}
+	return bins
+}
+
+//fillSmallestBin fills the bin found by findSmallestBin and adds the next contig to it.
+func fillSmallestBin(bins map[int][]Fasta, genome []Fasta) map[int][]Fasta {
+
+	for i := len(bins); i < len(genome); i++ {
+		b := findSmallestBin(bins)
+		bins[b] = append(bins[b], genome[i])
+	}
+
+	return bins
+}
+
+//findSmallestBin finds the bin with the least sequence and returns which bin it is so that it can be filled with the next contig.
```

Comment:
More language...
Instead of "least sequence" could this be called the "smallest contig"?
Can also talk more about this in code review.

Reply:
This is a good suggestion, but it can contain more than just a single contig it will take as many as it needs to so its not the smallest contig necessarily